src/System.Private.CoreLib/shared/System/MemoryExtensions.Fast.cs

```diff
@@ -219,11 +219,14 @@ public static int LastIndexOf(this ReadOnlySpan<char> span, ReadOnlySpan<char> v
         /// <param name="source">The source span.</param>
         /// <param name="destination">The destination span which contains the transformed characters.</param>
         /// <param name="culture">An object that supplies culture-specific casing rules.</param>
-        /// <remarks>If the source and destinations overlap, this method behaves as if the original values are in
-        /// a temporary location before the destination is overwritten. If <paramref name="culture"/> is null, <see cref="System.Globalization.CultureInfo.CurrentCulture"/> will be used.</remarks>
+        /// <remarks>If the source and destinations overlap, this method throws InvalidOperationException. 
+        /// If <paramref name="culture"/> is null, <see cref="System.Globalization.CultureInfo.CurrentCulture"/> will be used.</remarks>
         /// <returns>The number of characters written into the destination span. If the destination is too small, returns -1.</returns>
         public static int ToLower(this ReadOnlySpan<char> source, Span<char> destination, CultureInfo? culture)
         {
+            if (source.Overlaps(destination))
+               ThrowHelper.ThrowInvalidOperationException(ExceptionResource.NotSupported_SpanOverlappedOperation);
```

Comment:
This does not need to use ThrowHelper. ThrowHelper should be used only when it makes a perf difference (e.g. generic method or tiny perf criticial methods). It is not the case.

Reply:
I thought it was a "standard" way to throw in all codebases...also in this file there are only helper calls.
I'll update with simple throw exception.

Reply:
It is far from standard. In CoreLib, there are ~6000 calls to `throw new ...` and ~1000 calls to `ThrowHelper.Something`. The ratio is even more significant in CoreFX.

The other places in this file that use ThrowHelper fall into "generic method or tiny perf criticial methods" category.

Reply:
Uh...thought more... thanks for details